src/main/java/org/opensearch/security/dlic/rest/validation/PasswordValidator.java

```diff
@@ -0,0 +1,185 @@
+import java.util.regex.Pattern;
+
+import com.google.common.collect.ImmutableList;
+import com.nulabinc.zxcvbn.Strength;
+import com.nulabinc.zxcvbn.Zxcvbn;
+import com.nulabinc.zxcvbn.matchers.Match;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.common.Strings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.security.dlic.rest.validation.AbstractConfigurationValidator.ErrorType;
+
+import static org.opensearch.security.support.ConfigConstants.SECURITY_RESTAPI_PASSWORD_MIN_LENGTH;
+import static org.opensearch.security.support.ConfigConstants.SECURITY_RESTAPI_PASSWORD_SCORE_BASED_VALIDATION_STRENGTH;
+import static org.opensearch.security.support.ConfigConstants.SECURITY_RESTAPI_PASSWORD_VALIDATION_REGEX;
+
+public class PasswordValidator {
+
+    private static final int MAX_LENGTH = 100;
+
+    /**
+     * Checks a username similarity and a password
+     * names and passwords like:
+     *  - some_user_name/456Some_uSer_Name_1234
+     *  - some_user_name/some_user_name_Ydfge
+     *  - some_user_name/eman_resu_emos
+     *  are similar
+     * "user_inputs" - is a default dictionary zxcvbn creates for checking similarity
+     */
+    private final static Predicate<Match> USERNAME_SIMILARITY_CHECK = m ->
+            m.pattern == com.nulabinc.zxcvbn.Pattern.Dictionary && "user_inputs".equals(m.dictionaryName);
+
+    private final Logger logger = LogManager.getLogger(this.getClass());
+
+    private final int minPasswordLength;
+
+    private final Pattern passowrdRegexpPattern;
+
+    private final ScoreStrength scoreStrength;
+
+    private final Zxcvbn zxcvbn;
+
+    private PasswordValidator(final int minPasswordLength,
+                              final Pattern passowrdRegexpPattern,
+                              final ScoreStrength scoreStrength) {
+        this.minPasswordLength = minPasswordLength;
+        this.passowrdRegexpPattern = passowrdRegexpPattern;
+        this.scoreStrength = scoreStrength;
+        this.zxcvbn = new Zxcvbn();
+    }
+
+    public static PasswordValidator of(final Settings settings) {
+        final String passwordRegex = settings.get(SECURITY_RESTAPI_PASSWORD_VALIDATION_REGEX, null);
+        final ScoreStrength scoreStrength = ScoreStrength.fromConfiguration(
+                    settings.get(SECURITY_RESTAPI_PASSWORD_SCORE_BASED_VALIDATION_STRENGTH, ScoreStrength.STRONG.name())
+            );
+        final int minPasswordLength = settings.getAsInt(SECURITY_RESTAPI_PASSWORD_MIN_LENGTH, -1);
+        return new PasswordValidator(
+                minPasswordLength,
+                !Strings.isNullOrEmpty(passwordRegex) ? Pattern.compile(String.format("^%s$", passwordRegex)) : null,
+                scoreStrength);
+    }
+
+    ErrorType validate(final String username, final String password) {
+        if (minPasswordLength > 0 && password.length() < minPasswordLength) {
+            logger.debug(
```

Comment:
These seems like well written error messages seem valuable, could we return this messages instead to be visible to the user. If we did that, we could remove the password_validation_error_message (and related) value that is in the config.  What do you think?